### PR TITLE
♻️Refactor Navbar And Router Usage

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Adjust Router Config for Navbar
2. Use Aurelia Router To Generate The Navbar Entries
3. Add routerNavigate Method
4. Remove navigateToInspect, navigateToDesigner and navigateToThink Methods
5. Add navigate Method
6. Use routerNavigate instead of navigateToRoute

PR: #1275 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

- check whether the studio works as expected 

note: you should specially check for the navigation between the think design inspect and their child views (xml, diff, detail, inspect-correlation, heatmap, dashboard)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -147,17 +147,17 @@ export class App {
       },
       {
         route: ['processdef', 'processdef/:diagramName?'],
-        title: 'Process Definition List',
+        title: 'Think',
         name: 'processdef-list',
```
Comment (Vyperus) [inline src/app.ts]: You can change the name here as well. Otherwise I will do it in my PR anyway.

File: src/modules/navbar/navbar.ts [hunk 2]
```diff
@@ -130,15 +131,40 @@ export class NavBar {
...
-    this._router.navigateToRoute('think', {
-      view: 'diagram-list',
-      diagramName: this.activeDiagram ? this.activeDiagram.name : undefined,
-      solutionUri: this.activeSolutionEntry ? this.activeSolutionEntry.uri : undefined,
-    });
+  public navigate(navModel: NavModel): void {
+    switch (navModel.config.name) {
+      case 'think':
+        this.routerNavigate(navModel.config.name, this.thinkView);
+        break;
```
Comment (SteffenKn) [inline src/modules/navbar/navbar.ts]: Add some spacing
```suggestion

        break;
```

File: src/modules/navbar/navbar.ts [hunk 3]
```diff
@@ -130,15 +131,40 @@ export class NavBar {
...
+          return;
+        }
+
+        const designIsActive: boolean = navModel.isActive === true;
+        if (designIsActive) {
+          return;
+        }
+
+        this.routerNavigate(navModel.config.name, this.designView);
+        break;
```
Comment (SteffenKn) [inline src/modules/navbar/navbar.ts]: Add some spacing
```suggestion

        break;
```

File: src/modules/navbar/navbar.ts [hunk 4]
```diff
@@ -130,15 +131,40 @@ export class NavBar {
...
+      case 'inspect':
+        const inspectIsActive: boolean = navModel.isActive === true;
+        if (inspectIsActive) {
+          return;
+        }
+
+        this.routerNavigate(navModel.config.name, this.inspectView);
+        break;
```
Comment (SteffenKn) [inline src/modules/navbar/navbar.ts]: Add some spacing
```suggestion

        break;
```

**Comment by SteffenKn** [comment]
Please always add a step for step description for everything that needs to be tested in the 'How can others test the changes?'-section.
That way others can test like you did and they also might notice things you haven't tested.

**Comment by alexanderkasten** [comment]
> Please always add a (e.g. step by step) description for everything that needs to be tested in the 'How can others test the changes?'-section.
> That way others can test like you did and they also might notice things you haven't tested.

done

**Comment by SteffenKn** [comment]
> done

This: 
> - check whether the studio works as excepted

is not helpful at all.

**Comment by alexanderkasten** [comment]
> > done
> 
> This:
> 
> > * check whether the studio works as excepted
> 
> is not helpful at all.

> note: you should specially check for the navigation between the think design inspect and their child views (xml, diff, detail, inspect-correlation, heatmap, dashboard)

I think thats helpful enough

**Pull request merged**